nixos/modules/services/backup/tarsnap.nix

```diff
@@ -242,9 +254,11 @@ in
 
     systemd.services."tarsnap@" = {
       description = "Tarsnap archive '%i'";
-      requires    = [ "network.target" ];
+      requires    = [ "network-online.target" ];
+      after       = [ "network-online.target" ];
 
-      path = [ pkgs.tarsnap pkgs.coreutils ];
+      path = [ pkgs.tarsnap pkgs.coreutils pkgs.iputils ];
+      preStart = "/run/current-system/sw/bin/sh -c 'while ! ping -q -c 1 betatest-server.tarsnap.com &> /dev/null; do true; done'";
```

Comment:
I am unaffected by the backup-will-eventually-succeed-but-not-right-now problem and failed to consider it, but I can see how preStart might be warranted, esp. for people who backup rarely and so could end up not backing up for a long time if the service is triggered at bootup. It is unfortunate that network-online.target is underspecified on NixOS, but that seems to be a more general problem.